packages/editor/src/components/collab-sidebar/index.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * WordPress dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { comment as commentIcon } from '@wordpress/icons';
+
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import PluginSidebar from '../plugin-sidebar';
+import { collabSidebarName } from './constants';
+import { Comments } from './comments';
+import { AddComment } from './add-comment';
+
+const isBlockCommentExperimentEnabled =
+	window?.__experimentalEnableBlockComment;
+
+const modifyBlockCommentAttributes = ( settings, name ) => {
+	if ( name && name?.includes( 'core/' ) ) {
+		if (
+			undefined === settings.attributes.blockCommentId ||
+			null === settings.attributes.blockCommentId
+		) {
+			settings.attributes = {
+				...settings.attributes,
+				blockCommentId: {
+					type: 'number',
+					default: 0,
+				},
+				showCommentBoard: {
+					type: 'boolean',
+					default: false,
+				},
+			};
+		}
+	}
+	return settings;
+};
+
+// Apply the filter to all core blocks
+addFilter(
+	'blocks.registerBlockType',
+	'block-comment/modify-core-block-attributes',
+	modifyBlockCommentAttributes
+);
+
+/**
+ * Renders the Collab sidebar.
+ */
+export default function CollabSidebar() {
+	const [ threads, setThreads ] = useState( [] );
```

Comment:
Could use lazy init, that way we'll only create one array, and not one on every re-render.

```suggestion
	const [ threads, setThreads ] = useState( () => [] );
```